extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -392,12 +395,17 @@ export async function newDeployment(
 
   const getContentRecords = new Promise<void>(async (resolve, reject) => {
     try {
-      const response = await api.contentRecords.getAll();
+      const response = await api.contentRecords.getAll({ recursive: true });
       const contentRecordList = response.data;
       // Note.. we want all of the contentRecord filenames regardless if they are valid or not.
-      contentRecordNames = contentRecordList.map(
-        (contentRecord) => contentRecord.deploymentName,
-      );
+      contentRecordList.forEach((contentRecord) => {
+        let existingList = contentRecordNames.get(contentRecord.projectDir);
+        if (existingList === undefined) {
+          existingList = [];
+        }
+        existingList.push(contentRecord.deploymentName);
+        contentRecordNames.set(contentRecord.projectDir, existingList);
+      });
```

Comment:
Populate the map to be keyed by a projectDir and then contain an array of entrypoints.